x-pack/apm-server/sampling/eventstorage/storage.go

```diff
@@ -206,9 +209,14 @@ func (rw *ReadWriter) writeEntry(e *badger.Entry, opts WriterOpts) error {
 	if err := rw.Flush(opts.StorageLimitInBytes); err != nil {
 		return err
 	}
+	rw.pendingSize += entrySize
```

Comment:
This doesn't seem right. I think we'll need to track the amount of bytes buffered across all writers. Maybe this should be an `atomic.Uint64` on the `Storage` type?